packages/design-system/src/components/InputCode/SingleInput.tsx

```diff
@@ -0,0 +1,114 @@
+import { RefObject, ClipboardEvent } from "react";
+import classNames from "classnames";
+
+import Input from "@components/Input";
+
+import { SingleInputProps } from "./InputCode.interface";
+import { useInputCode } from "./useInputCode";
+import {
+  characterCode,
+  getInputMode,
+  getPattern,
+  isValid,
+  setElementValue,
+} from "./utils";
+
+export const SingleInput = ({
+  inputRefsArray,
+  dashesPositions,
+  onChange,
+  onFocus,
+  idx,
+  disabled,
+  type = "numbers",
+  inputClassName,
+  isError,
+  element,
+  ariaLabel,
+}: SingleInputProps) => {
+  const { inputsWithDashBefore, validate, changeInputFocus } = useInputCode(
+    type,
+    inputRefsArray,
+    dashesPositions
+  );
+
+  const handleOnChange = (
+    element: RefObject<HTMLInputElement>,
+    idx: number,
+    isBackAction?: boolean
+  ) => {
+    if (!!validate(element)) {
+      onChange && onChange();
+      changeInputFocus(idx, !!isBackAction);
+    }
+  };
```

Comment:
```suggestion
const onChangeFunction = onChange && onChange();

  const handleOnChange = (
    element: RefObject<HTMLInputElement>,
    idx: number,
    isBackAction?: boolean
  ) => {
    if (!!validate(element)) {
      onChangeFunction();
      changeInputFocus(idx, !!isBackAction);
    }
  };
```
and further down the code